pkg/scheduler/framework/plugins/podtopologyspread/scoring.go

```diff
@@ -174,13 +174,13 @@ func (pl *PodTopologySpread) PreScore(
 func (pl *PodTopologySpread) Score(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
 	nodeInfo, err := pl.sharedLister.NodeInfos().Get(nodeName)
 	if err != nil || nodeInfo.Node() == nil {
-		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("getting node %q from Snapshot: %v, node is nil: %v", nodeName, err, nodeInfo.Node() == nil))
+		return 0, framework.AsStatus(fmt.Errorf("getting node %q from Snapshot: %w, node is nil: %v", nodeName, err, nodeInfo.Node() == nil))
```

Comment:
`%w` should be at the end of the message.

IIRC, `nodeInfo.NodeI()` can no longer be nil, as we do the check when we create the Snapshot. Would you mind verifying and removing that check?

Reply:
There is a check here
 https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/podtopologyspread/scoring.go#L142

Reply:
Done

Reply:
/hold

The source of truth is actually the scheduler cache https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/internal/cache/cache.go#L201

And it looks like there might actually be a bug there were we consider some Nodes that are nil. I'm evaluating.